### PR TITLE
plan: make agg's cost formula more reasonable.

### PR DESCRIPTION
**Pull request by winoros**
Based on real benchmark result.
PTAL @XuHuaiyu @zz-jason

### CODE REVIEW

File: plan/task.go [hunk 1]
```diff
@@ -521,9 +528,15 @@ func (p *PhysicalStreamAgg) attach2Task(tasks ...task) task {
 		t = finishCopTask(p.ctx, cop)
 		attachPlan2Task(finalAgg, t)
 		t.addCost(t.count() * cpuFactor)
+		if hasDistinct {
```
Comment (XuHuaiyu) [inline plan/task.go]: can this check be put on line 541,
thus the check in lint 537 can be removed.

File: plan/task.go [hunk 2]
```diff
@@ -507,6 +507,13 @@ func (p *PhysicalStreamAgg) attach2Task(tasks ...task) task {
 		return invalidTask
 	}
 	t := tasks[0].copy()
+	hasDistinct := false
+	for _, aggFunc := range p.AggFuncs {
```
Comment (XuHuaiyu) [inline plan/task.go]: can this be moved to line 541

File: plan/task.go [hunk 3]
```diff
@@ -535,6 +548,13 @@ func (p *PhysicalHashAgg) attach2Task(tasks ...task) task {
 	}
 	cardinality := p.StatsInfo().count
 	task := tasks[0].copy()
+	hasDistinct := false
+	for _, aggFunc := range p.AggFuncs {
```
Comment (XuHuaiyu) [inline plan/task.go]: ditto

File: plan/task.go [hunk 4]
```diff
@@ -549,10 +569,16 @@ func (p *PhysicalHashAgg) attach2Task(tasks ...task) task {
 		}
 		task = finishCopTask(p.ctx, cop)
 		attachPlan2Task(finalAgg, task)
-		task.addCost(task.count()*cpuFactor + cardinality*hashAggMemFactor)
+		task.addCost(task.count()*cpuFactor*hashAggFactor + cardinality*createAggCtxFactor)
```
Comment (XuHuaiyu) [inline plan/task.go]: ditto

File: plan/task.go [hunk 5]
```diff
@@ -520,10 +520,19 @@ func (p *PhysicalStreamAgg) attach2Task(tasks ...task) task {
...
 		attachPlan2Task(p, t)
-		t.addCost(t.count() * cpuFactor)
+	}
+	hasDistinct := false
+	for _, aggFunc := range p.AggFuncs {
+		if aggFunc.HasDistinct {
+			hasDistinct = true
+			break
+		}
+	}
```
Comment (zimulala) [inline plan/task.go]: Could this be extracted into a function？

**Pull request merged**